web/vibe/web/rest.d

```diff
@@ -469,23 +469,24 @@ URLRouter registerRestInterface(TImpl)(URLRouter router, TImpl instance, string
 HTTPServerRequestDelegate serveRestJSClient(I)(RestInterfaceSettings settings)
 	if (is(I == interface))
 {
+	import std.datetime.systime : SysTime;
 	import std.digest.md : md5Of;
 	import std.digest : toHexString;
 	import std.array : appender;
 
+	import vibe.http.fileserver : ETag, handleCache;
+
 	auto app = appender!string();
 	generateRestJSClient!I(app, settings);
 	auto hash = app.data.md5Of.toHexString.idup;
+	ETag tag;
+	tag.tag = hash;
```

Comment:
Better to use a constructor/builder that takes just `app.data` as we then have one place to change/optimize the MD5 computation.

Reply:
this is also just an arbitrary decision using the md5 algorithm on the contents of the file in this case. I don't really know how to better abstract something which should be an opaque string without meaning because `.md5Of.toHexString.idup` is already pretty good to use if you got the data imo

Of course the user can pick between any algorithm or use crc32 or stuff like that

Reply:
Again my ideas was to add sth. like a generic `fromBytes` to `ETag` as this looks like the wrong place to decide about the hashing algorithm.